### PR TITLE
[SD-3760] Revisit ingest providers conversion of preformatted to html

### PR DESCRIPTION
**Pull request by marwoodandrew**
Remove the source specific Feeding Services and make specific Feed Parser's, for AFP and DPA.
(The delivery mechanism should not be a function of the feed format where possible)

ANPA and DPA parser's now markup the body_html.
 
Add AP NITF sample, test and make it work.

The PR is dependent on a PR to the Superdesk repo.

### CODE REVIEW

**Comment by sivakuna-aap** [comment]
:+1: :100: :purple_heart: :green_heart: :heart: :wink:

**Comment by mdhaman** [comment]
:+1:

File: superdesk/io/feed_parsers/dpa_iptc7901.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    def parse(self, file_path, provider=None):
+        item = super().parse(file_path, provider)
+        item = self.dpa_derive_dateline(item)
+        # Markup the text and set the content type
+        item['body_html'] = '<p>' + item['body_html'].replace('\r\n', ' ').replace('\n', '</p><p>') + '</p>'
+        item[ITEM_TYPE] = CONTENT_TYPE.TEXT
+        return item
+
+    def dpa_derive_dateline(self, item):
```
Comment (mugurrus) [inline superdesk/io/feed_parsers/dpa_iptc7901.py]: A unit test for this would be good.

**Comment by sivakuna-aap** [comment]
@marwoodandrew Sorry for the late comment.

Can we make feed parser optional? 

At the moment there are likely chances for failure of `update_ingest` if Feeding Service doesn't return items. Reason for asking this few days back there was an issue raised by @actionless that there's no option to create a Ingest Provider without a Feed Parser. 

Can we modify `update_ingest` so that it doesn't fail if feeding service doesn't return any items and remove the hard coding here? https://github.com/superdesk/superdesk/blob/master/client/app/scripts/superdesk-ingest/views/settings/ingest-sources-content.html#L62

**Comment by marwoodandrew** [comment]
OK, might be Mondays problem!

**Comment by sivakuna-aap** [comment]
agreed :smile:

**Comment by petrjasek** [comment]
:+1:

**Comment by akintolga** [comment]
:+1:

**Pull request merged**